### PR TITLE
handle dict preloading for agg

### PR DESCRIPTION
**Pull request by PSeitz**
handle dict preloading for agg
handle cardinality for fields

### CODE REVIEW

File: quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs [hunk 1]
```diff
@@ -147,14 +147,7 @@ impl DefaultDocMapperBuilder {
         let field_entries = root_field_mapping.compute_field_entries();
         let required_fast_fields: Vec<FieldPath> = field_entries
             .into_iter()
-            .filter(|(_, field_type)| match field_type {
-                FieldType::U64(options)
-                | FieldType::I64(options)
-                | FieldType::F64(options)
-                | FieldType::Date(options) => options.is_fast(),
-                FieldType::Bytes(option) => option.is_fast(),
-                _ => false,
```
Comment (fulmicoton) [inline quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs:156]: you mentionned there was a bug in tantiv? Is it solved?

File: quickwit-search/src/leaf.rs [hunk 2]
```diff
@@ -121,23 +125,83 @@ pub(crate) async fn open_index(
 ///
 /// The downloaded data depends on the query (which term's posting list is required,
 /// are position required too), and the collector.
+///
+/// * `query` - query is used to extract the terms and the fields which will be loaded from the
```
Comment (fulmicoton) [inline quickwit-search/src/leaf.rs:129]: in which case do we load a field due to the query? what does  a field mean here? Is it a fast field?

File: quickwit-search/src/leaf.rs [hunk 3]
```diff
@@ -121,23 +125,83 @@ pub(crate) async fn open_index(
...
+                let dict = inverted_index.terms();
+                dict.warm_up_dictionary().await
+            });
+        }
+    }
+    try_join_all(warm_up_futures).await?;
     Ok(())
 }
 
+pub fn get_fastfield_cardinality(field_type: &FieldType) -> Option<Cardinality> {
```
Comment (fulmicoton) [inline quickwit-search/src/leaf.rs:192]: does this need to be public?

Comment (fulmicoton) [inline quickwit-search/src/leaf.rs:192]: Thanks for thinking about the name. (calling it fastfield_cardinality.)
The notion of cardinality is indeed ambiguous.

A single valued bytes field in quickwit has a multivalued fast field cardinality.

Maybe adding a comment would be a good addition. 
(and make the function private)

File: quickwit-search/src/leaf.rs [hunk 4]
```diff
@@ -158,14 +222,42 @@ async fn warm_up_fastfields(
...
+                    fast_field_name, field_entry
+                )
+            })?;
+
+        fast_fields.push((fast_field, cardinality));
     }
 
-    let mut warm_up_futures = Vec::new();
-    for field in fast_fields {
+    type SendebleFuture = dyn Future<Output = Result<OwnedBytes, AsyncIoError>> + Send;
```
Comment (fulmicoton) [inline quickwit-search/src/leaf.rs:236]: Sendable?

**Review by fulmicoton — APPROVED** [review]
see comments inline

File: quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs (hunk 1 above)
Comment (PSeitz) [reply quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs:156]: will be fixed with https://github.com/quickwit-oss/tantivy/pull/1346

File: quickwit-search/src/leaf.rs (hunk 2 above)
Comment (PSeitz) [reply quickwit-search/src/leaf.rs:129]: I just commented on the function to document the behaviour. field is a tantivy `Field` here.

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**